auth0/src/main/java/com/auth0/android/request/DefaultClient.kt

```diff
@@ -14,12 +15,24 @@ import java.util.concurrent.TimeUnit
 
 /**
  * Default implementation of a Networking Client.
+ *
+ * @param connectTimeout the connection timeout, in seconds, to use when executing requests. Default is ten seconds.
+ * @param readTimeout the read timeout, in seconds, to use when executing requests. Default is ten seconds.
+ * @param defaultHeaders any headers that should be sent on all requests. If a specific request specifies a header with the same key as any header in the default headers, the header specified on the request will take precedence. Default is an empty map.
+ * @param enableLogging whether HTTP request and response info should be logged. This should only be set to `true` for debugging purposes in non-production environments, as sensitive information is included in the logs. Defaults to `false`.
  */
-public class DefaultClient() : NetworkingClient {
+public class DefaultClient(
+        private val connectTimeout: Long = DEFAULT_TIMEOUT_SECONDS,
+        private val readTimeout: Long = DEFAULT_TIMEOUT_SECONDS,
+        private val defaultHeaders: Map<String, String> = mapOf(),
+        private val enableLogging: Boolean = false
+) : NetworkingClient {
 
     //TODO: receive this via internal constructor parameters
     private val gson: Gson = GsonProvider.buildGson()
-    private var client: OkHttpClient
+
+    @get:VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal var okHttpClient: OkHttpClient
```

Comment:
Please change this into a `val` instead. I'd also add this on the line below to make the setter private. Unless there's a reason to open that?

```
private set
```

Reply:
Updated to val. No need (or ability) to add `private set` as it is now a val.